src/Symfony/Component/Mime/Part/TextPart.php

```diff
@@ -207,7 +208,18 @@ private function getEncoder(): ContentEncoderInterface
             return self::$encoders[$this->encoding] ??= new QpContentEncoder();
         }
 
-        return self::$encoders[$this->encoding] ??= new Base64ContentEncoder();
+        if ('base64' === $this->encoding) {
+            return self::$encoders[$this->encoding] ??= new Base64ContentEncoder();
+        }
+
+        return self::$encoders[$this->encoding];
+    }
+
+    public function addEncoder(string $name, ContentEncoderInterface $encoder): static
+    {
+        self::$encoders[$name] = $encoder;
```

Comment:
Would this line also allow me to change the three default encoders? If so, is this something we want to allow?

Reply:
It would yeah, but I agree it's probably better if the user explicitly has to add a new encoder when they want to modify the behaviour of existing ones.